models/opnilog/opnilog_model.py

```diff
@@ -301,28 +299,19 @@ def __init__(
         self.tokenizer = tokenizer
 
     def get_sample_weights(self):
-        def changeTokenToCount(token, dictInfo):
-            if token == 0:
-                return 0
-            else:
-                return dictInfo[token]
-
-        d = self.c(self.padded_data)
-        data_token_idx_df = pd.DataFrame(d)
-        storeColumnInfo = defaultdict(dict)
-        cnt = 0
-        for column in range(self.pad_len):
-            val_cnt = pd.value_counts(data_token_idx_df.iloc[:, column])
-            storeColumnInfo[column] = val_cnt.to_dict()
-            data_token_idx_df.iloc[:, column] = data_token_idx_df.iloc[:, column].apply(
-                lambda x: changeTokenToCount(x, storeColumnInfo[column])
-            )
-        #         weights = minmax_scale(np.divide(np.ones(data_token_idx_df.shape[0]),
-        #                                          data_token_idx_df.sum(axis=1)),
-        #                                feature_range=(0.005, 0.995))
-        weights = 1 - minmax_scale(
-            data_token_idx_df.sum(axis=1), feature_range=(0.0, 0.75)
-        )
+        """
+        This function aims to address the case the data is very imbalanced, and make the data less imbalanced.
+        """
+        unique_sample_counter = defaultdict(int)
+        str_data = [str(l) for l in self.data]
+        for s in str_data:
```

Comment:
this is inefficient. can loop self.data and can str() inside the loop to avoid unnecessary storage for `str_data`

Reply:
Not really, `str_data` is used again at line 314.

Reply:
you can loop `self.data`. saving storage >>> saving compute here
306-308:
```
for s in self.data:
   unique_sample_counter[str(s)] += 1
```
314: `weights = np.array([unique_sample_counter[str(s)] for s in self.data])`

Reply:
saving storage is not >>> compute here bc `self.data` contains number of valid tokens not the actual string. But if you insist, sure.

Reply:
okay then why not directly modify type of self.data?